### PR TITLE
Add metrics to vl3 NSE

### PR DESCRIPTION
**Pull request by ondrej-fabry**
Original issue: https://github.com/cisco-app-networking/cnns-prj/issues/83

Related PRs:
- https://github.com/cisco-app-networking/wcm-system/pull/110 

- [x] added vl3 NSE metrics
- [x] added gRPC metrics (https://github.com/grpc-ecosystem/go-grpc-prometheus)

### CODE REVIEW

File: cmd/vl3-nse/nse.go [hunk 1]
```diff
@@ -93,6 +101,13 @@ func main() {
 	mainFlags := &Flags{}
 	mainFlags.Process()
 
+	// Metrics
+	metricsPort := os.Getenv(metricsPortEnv)
```
Comment (tiswanso) [inline cmd/vl3-nse/nse.go]: I think you want to check for metricsPort == "" and default it.

File: deployments/helm/vl3/templates/vl3-nse-ucnf.tpl [hunk 2]
```diff
@@ -114,5 +118,22 @@ spec:
   ports:
     - name: monitoring
       port: {{ .Values.vppMetricsPort }}
+      targetPort: monitoring-vpp
+      protocol: TCP
+---
+apiVersion: v1
+kind: Service
```
Comment (tiswanso) [inline deployments/helm/vl3/templates/vl3-nse-ucnf.tpl:125]: why not just add to the existing service's port list?

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+			Name:      "received_conn_requests",
+			Help:      "Total number of received connection requests from vL3 NSE",
+		})
+	PerormedConnRequests = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "vl3_nse",
+			Name:      "performed_conn_requests",
+			Help:      "Total number of performed connection requests to vL3 NSE",
+		})
+	// TODO: define more metrics
```
Comment (ondrej-fabry) [inline pkg/metrics/metrics.go]: Any ideas what other metrics might be important here?

**Review by tiswanso — COMMENTED** [review]
Looks good ... only question is whether the grpc_prometheus interceptors require any connectivity to a prometheus installation or if it's just adding some standard metrics to the available collection from the NSE's prometheus metrics server?

**Comment by ondrej-fabry** [comment]
@tiswanso Yes, those gRPC metrics are same as the ones defined for vl3 NSE in metrics.go, since we used the default prometheus registerer. The gRPC client metrics are
-  registered here: https://github.com/grpc-ecosystem/go-grpc-prometheus/blob/master/client.go
- defined here: https://github.com/grpc-ecosystem/go-grpc-prometheus/blob/master/client_metrics.go

I actually recommend using these [gRPC server/client metrics](https://github.com/grpc-ecosystem/go-grpc-prometheus#metrics) in every component that uses gRPC and exports metrics. It can actually even do [histograms for gRPCs calls](https://github.com/grpc-ecosystem/go-grpc-prometheus#histograms) (needs to be enabled).

**Review by tiswanso — APPROVED** [review]
misspelled "performed" in "PerormedConnRequests" but we can fix it in a follow-up.

**Pull request merged**